optuna/storages/_rdb/alembic/versions/v3.0.0.c.py

```diff
@@ -0,0 +1,123 @@
+"""Add intermediate_value_type column to represent +inf and -inf
+
+Revision ID: v3.0.0.c
+Revises: v3.0.0.b
+Create Date: 2022-05-16 17:17:28.810792
+
+"""
+import enum
+
+import numpy as np
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import orm
+
+
+# revision identifiers, used by Alembic.
+revision = "v3.0.0.c"
+down_revision = "v3.0.0.b"
+branch_labels = None
+depends_on = None
+
+
+BaseModel = declarative_base()
+RDB_MAX_FLOAT = np.finfo(np.float32).max
+RDB_MIN_FLOAT = np.finfo(np.float32).min
+
+
+class IntermediateValueModel(BaseModel):
+    class FloatTypeEnum(enum.Enum):
+        FINITE_OR_NAN = 1
+        INF_POS = 2
+        INF_NEG = 3
+
+    __tablename__ = "trial_intermediate_values"
+    trial_intermediate_value_id = sa.Column(sa.Integer, primary_key=True)
+    intermediate_value = sa.Column(sa.Float, nullable=True)
+    intermediate_value_type = sa.Column(sa.Enum(FloatTypeEnum), nullable=False)
+
+
+def upgrade():
+    # MySQL and PostgreSQL supports DEFAULT clause like 'ALTER TABLE <tbl_name>
+    # ADD COLUMN <col_name> ... DEFAULT "FINITE_OR_NAN"', but seemingly Alembic
+    # does not support such a SQL statement. So first add a column with schema-level
+    # default value setting, then remove it by `batch_op.alter_column()`.
+    with op.batch_alter_table("trial_intermediate_values") as batch_op:
+        batch_op.add_column(
+            sa.Column(
+                "intermediate_value_type",
+                sa.Enum("FINITE_OR_NAN", "INF_POS", "INF_NEG", name="floattypeenum"),
+                nullable=False,
+                server_default="FINITE_OR_NAN",
+            ),
+        )
+    with op.batch_alter_table("trial_intermediate_values") as batch_op:
+        batch_op.alter_column("intermediate_value_type", server_default=None)
```

Comment:
We actually need to execute these operations in separated batch (it means SQLite3 database recreated twice), otherwise `(sqlite3.IntegrityError) NOT NULL constraint failed: _alembic_tmp_trial_intermediate_values.intermediate_value_type` exception will be raised.